### PR TITLE
[stdlib] Conditionally compile swift_strto* wrappers

### PR DESCRIPTION
**Pull request by kastiglione**
Use conditional compilation to prevent unused function warnings for `swift_strto*` compat functions.

rdar://problem/70445001

### CODE REVIEW

File: stdlib/public/stubs/Stubs.cpp [hunk 1]
```diff
@@ -378,6 +378,28 @@ T _swift_strto(const char *nptr, char **endptr) {
   return ParsedValue;
 }
 
+#if defined(__OpenBSD__) || defined(_WIN32) || defined(__CYGWIN__) || defined(__HAIKU__)
+#define need_swift_strtod_l
```
Comment (kastiglione) [inline stdlib/public/stubs/Stubs.cpp:382]: I moved this block before the functions, and added these `need_swift_strto*` "variables". Each function is then gated by the definition of the appropriate variable.

Comment (compnerd) [reply stdlib/public/stubs/Stubs.cpp:382]: The upfront configuration makes sense.  The only thing I find a bit difficult to read is the difference between the `need_*` and `strto*` renaming.  I don't really have much of a good solution to that other than perhaps use upper case macro names for the `NEED_*`?

**Review by compnerd — APPROVED** [review]
Logically this change is fine.  The stylistic thing that can be improved is well, stylistic, and I'm not convinced my (weak) preference here is any better than the current proposal.

File: stdlib/public/stubs/Stubs.cpp (hunk 1 above)
Comment (kastiglione) [reply stdlib/public/stubs/Stubs.cpp:382]: Uppercase works for me. I wondered about that when doing it initially.

Comment (kastiglione) [reply stdlib/public/stubs/Stubs.cpp]: updated

**Comment by kastiglione** [comment]
@swift-ci smoke test

**Comment by kastiglione** [comment]
@swift-ci smoke test

**Comment by kastiglione** [comment]
@swift-ci test Windows Platform

**Comment by kastiglione** [comment]
@swift-ci test Windows Platform

**Comment by kastiglione** [comment]
@swift-ci test Windows Platform

**Pull request merged**